### PR TITLE
fix: should run validation on submit and clear error if a valid  value is entered

### PR DESCRIPTION
**Pull request by vikaskumar89**
This PR fixes issue https://github.com/TanStack/form/issues/490 and https://github.com/TanStack/form/issues/486.

## What is changed
- run onChange/onBlur validation on form submit . If the validation fails, form is disabled
- clear the onSubmit error If a valid value is entered.

PR-form-1

### CODE REVIEW

**Comment by vikaskumar89** [comment]
> Is there anything currently preventing this from being merged?

Maintainer is very slow. cc. @tannerlinsley

**Comment by eastack** [comment]
I'm not sure when the maintainer will merge this, but is there any other way we can work around these issues temporarily?

**Comment by crutchcorn** [comment]
Hey all, maintainer here.

@vikaskumar89 I will need you to please start being patient with me. On top of TanStack Form not being 1.x yet (in which case bug fixes will be merged much faster), I also work on various other OSS projects, am trying to finish writing a book, am oftentimes busy with personal issues, and to top it off am not really getting paid for this work (minus a few generous donors albeit for relatively little amounts).

Please do not tag Tanner directly, and know that I will eventually come back to things here.

There is a reason I am "slow" :)

That all said, I will be reviewing this shortly.

File: packages/form-core/src/FieldApi.ts [hunk 1]
```diff
@@ -332,10 +332,15 @@ export class FieldApi<
   validateSync = (value = this.state.value, cause: ValidationCause) => {
     const { onChange, onBlur } = this.options
     const validate =
-      cause === 'submit' ? undefined : cause === 'change' ? onChange : onBlur
+      cause === 'submit'
+        ? onChange
+          ? onChange
+          : onBlur
```
Comment (crutchcorn) [inline packages/form-core/src/FieldApi.ts:338]: This is not the correct backup to `cause === "submit" && !onChange`

Instead, we need to run both `onChange` **and** `onBlur`.

I'll take over this PR to make the appropriate changes.

Comment (crutchcorn) [inline packages/form-core/src/FieldApi.ts:338]: This is not the correct backup to `cause === "submit" && !onChange`

Instead, we need to run both `onChange` **and** `onBlur`.

I'll take over this PR to make the appropriate changes.

**Review by crutchcorn — CHANGES_REQUESTED** [review]

**Review by crutchcorn — APPROVED** [review]
Issues with implementation have now been resolved.

In addition, I've added `<Field onSubmit={z.string().min(1)}/>` functionality, which was missing previously.

I also added more tests than before.

**Pull request merged**